src/components/Chat.tsx

```diff
@@ -615,8 +600,13 @@ ${chatContent}`;
       }
 
       const editor = leaf.view.editor;
-      const cursor = editor.getCursor();
-      editor.replaceRange(message, cursor);
+      const cursorFrom = editor.getCursor("from");
+      const cursorTo = editor.getCursor("to");
+      if (replace) {
+        editor.replaceRange(message, cursorFrom, cursorTo);
+      } else {
+        editor.replaceRange(message, cursorTo);
```

Comment:
change to cursorTo for insertion so when user selects text before insertion, it will add the text to the end of the selection, which is more intuitive.